draft-gpew-priv-ppm.md

```diff
@@ -969,11 +959,27 @@ prevent such replay attacks, this specification requires the aggregators to
 detect and filter out replayed reports.
 
 To detect replay attacks, each aggregator keeps track of the set of nonces
-pertaining to reports that were previously aggregated for a given task. If the
-leader receives a report from a client whose nonce is in this set, it simply
-ignores it. A helper who receives an encrypted input share whose nonce is
-in this set replies to the leader with an error as described in
-{{aggregate-request}}.
+pertaining to reports that were previously aggregated for a given task. Before
+processing a report, an aggregator first checks if the report pertains to a
+batch that has already been "collected". This is the case if the aggregator is
+the leader and has completed a collect request for a batch containing the report
+(see {{pa-collect}}) or if the aggregator is a helper and has completed an
+aggregate-share request for a batch containing the report (see
+{{aggregate-share-request}}). It also checks if the report has been
+"aggregated". This is the case if a report with the same nonce was previously
+used in an aggregate request.
+
+* If the report has not yet been aggregated but is contained by a batch that has
+  been collected, then the aggregator MUST ignore it. This prevents additional
+  reports from being aggregated after their batch has already been collected.
+
+* If the report has been aggregated but has not been collected, then the
+  aggregator MUST ignore it. This prevents a report from being used more than
+  once in a batch.
+
+In particular, an aggregator only process a report if either it has been
+aggregated and collected or it has never been aggregated and is not contained by
+a batch that has been collected.
```

Comment:
This new requirement is pretty subtle to me. Can we drop an open issue here to further investigate if this is indeed the correct thing to do?

Reply:
I agree it's subtle. I'd be good with that outcome, although it was already discussed here in a different context: https://github.com/abetterinternet/ppm-specification/issues/183#issuecomment-1016978627.

Reply:
Let's at least leave an OPEN ISSUE referencing #183, then, since that's still open. (I will review #183 more carefully soon.)